model/host/hostutil.go

```diff
@@ -156,7 +156,38 @@ func initSystemCommand() string {
 	`
 }
 
-// FetchJasperCommands builds the command to download and extract the Jasper
+// FetchAndReinstallJasperCommand returns the command to fetch Jasper and
+// restart the service with the latest version.
+func (h *Host) FetchAndReinstallJasperCommand(config evergreen.JasperConfig) string {
+	return strings.Join([]string{
+		h.FetchJasperCommand(config),
+		h.ForceReinstallJasperCommand(config),
+	}, " && ")
+}
+
+// ForceReinstallJasperCommand returns the command to stop the Jasper service,
+// delete the current Jasper service configuration (if it exists), install the
+// new configuration, and restart the service.
+func (h *Host) ForceReinstallJasperCommand(config evergreen.JasperConfig) string {
+	port := config.Port
+	if config.Port == 0 {
+		port = evergreen.DefaultJasperPort
+	}
+	return h.jasperServiceCommand(config, "force-reinstall", fmt.Sprintf("--port=%d", port))
+}
+
+func (h *Host) jasperServiceCommand(config evergreen.JasperConfig, subCmd string, args ...string) string {
+	binaryPath := filepath.Join(h.Distro.CuratorDir, h.jasperExtractedFileName(config))
+	cmd := fmt.Sprintf("%s jasper service %s rpc %s", binaryPath, subCmd, strings.Join(args, " "))
+	// Jasper service commands need elevated privileges to execute. On Windows,
+	// this is assuming that the command is already being run by Administrator.
+	if !h.Distro.IsWindows() {
+		cmd = "sudo " + cmd
```

Comment:
we should probably have a "use sudo" option to jasper.Command and also use that here.

Reply:
The setup host job is the one actually running this command. I didn't make this a Jasper command because this string will have to be reused by user data to run this exact same command, but the command has to be put in the user data script as a string, not a Jasper SSH command.

I added a [PR to add sudo](https://github.com/mongodb/jasper/pull/79) since it seems useful, but won't be using it here.

Reply:
yay, this is done.